### DISCUSSION TITLE
About Dr.jit printf_async

### DISCUSSION THREAD

**Discussion by lesphere** [mitsuba-renderer/mitsuba3#974] [Q&A] [answered]
Hi,

I'm trying to print some debug info with dr.printf_async(), but I wonder where the output goes. 

This is the example code with no output:

```python
a = mi.Uint(1)
dr.printf_async('test %d\n', a)
```

**Comment by merlinND** [comment, marked as answer]
Hello @lesphere,

Your example does not generate a kernel, so the the async print instruction never gets executed. The following example prints as expected:

```python
a = mi.UInt32([1, 2, 3])
dr.printf_async('test %d\n', a)
b = dr.sqr(a)
dr.eval(b)
```

**Reply by wjakob** [reply]
Also, this will be printed on the console. It might not show up in a Jupyter notebook. We will fix this soon.

**Reply by merlinND** [reply]
Good point, I only tried it in a REPL in the console.

**Reply by lesphere** [reply]
It works. But if I run the code for the second time, there's no output unless I `print(b)`.
